platform/mbed_retarget.cpp

```diff
@@ -251,53 +252,50 @@ extern "C" FILEHANDLE PREFIX(_open)(const char* name, int openmode) {
         if (!path.exists()) {
             /* The first part of the filename (between first 2 '/') is not a
              * registered mount point in the namespace.
-             * Free file handle.
              */
-            filehandles[fh_i] = NULL;
-            errno = ENOENT;
-            return -1;
-        } else if (path.isFile()) {
+            goto fail;
```

Comment:
It looks like the fail block could be wrapped up in a seperate function and reused in more places?